### ISSUE TITLE
Scan for npm package-lock.json when scanning a NuGet solution/project

### ISSUE THREAD

**Issue by crotondo-dap** [Inedo/pgscan#27]
Hi,

within Issue #25 you added a support to scan for npm package-lock-json when scanning a NuGet solution.
Unfortunately, we do not want parts of this behavior.
I can't see a switch to control this behavior. If I call 'pgscan identify' on a solution it automatically takes npm-dependencies into account, even if I say the type is "nuget". Am I missing something here? Or is it not possible to control this?
Further, all package-lock-json files are read recursively. Until now we were able to give pgscan a specific path to a specific package-lock-json. We don't want every file to be scanned. It doesn't seem to give a switch here neither?

So maybe adding a switch would be an option - which controls if all package-lock.json files should be taken into account automatically or not. Or an additional parameter to give the specific path to the package-lock.json if one wants to do it in one call but with only one specific package-lock.json file.

Let me know if I am missing something and what you think about this suggestions.

**Comment by rhessinger** [comment]
Hi @crotondo-dap,

I just release 1.4.3 that implements the fix and now will not scan NPM dependencies when the type is specified.

**Comment by aunverdorben-dap** [comment]
Hi @rhessinger,

thank you for fixing this so quickly. Regarding this:

> Further, all package-lock-json files are read recursively. 

We've seen that package-lock.json files from npm packages consumed by our project are also getting processed when no specific package-lock.json is specified on the command line. We think that all package-lock.json files under \node_modules should be ignored. 

Do you agree or do you think these files should also be processed?

**Comment by rhessinger** [comment]
Hi @aunverdorben-dap,

That is a fair point to make.  My thought was that including the node_modules folder in the recursive search would allow us to include the child dependencies used by installed packages that were not marked as dependencies in the npm package.  But in my research and testing, I have found the package-lock.json at the root of the node_modules folder includes a subset of the data in the main package-lock.json.  So no extra information was added. Do your package-lock.json files under the node-modules folder have additional information the parent doesn't? Also, do your packages in that folder have package-lock.json outside of the root of that folder?

Looking at the [hidden lock file](https://docs.npmjs.com/cli/v9/configuring-npm/package-lock-json#hidden-lockfiles) documentation.  The information in that file *should* be redundant as it is only used to improve performance, but if there is manual change in the node_modules tree by something other than npm, then the lock file is ignored (and should probably be removed anyways).  I'm inclined to just exclude files from the node_modules folder as you suggest.

Thanks,
Rich
